src/app_service/service/token/async_tasks.nim

```diff
@@ -29,6 +26,71 @@ const getSupportedTokenList*: Task = proc(argEncoded: string) {.gcsafe, nimcall.
     }
     arg.finish(output)
 
+type
+  FetchTokensMarketValuesTaskArg = ref object of QObjectTaskArg
+    symbols: seq[string]
+    currency: string
+
+const fetchTokensMarketValuesTask*: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
+  let arg = decode[FetchTokensMarketValuesTaskArg](argEncoded)
+  var response: RpcResponse[JsonNode]
```

Comment:
An example of compact writing async tasks:

https://github.com/status-im/status-desktop/blob/master/src/app_service/service/wallet_account/async_tasks.nim#L13-L24